lib/broadway_kafka/brod_client.ex

```diff
@@ -110,18 +110,14 @@ defmodule BroadwayKafka.BrodClient do
   end
 
   @impl true
-  def stop_group_coordinator(nil) do
+  def disconnect(nil, client_id) do
+    :ok = :brod.stop_client(client_id)
     :ok
   end
 
-  def stop_group_coordinator(group_coordinator) do
-    ref = Process.monitor(group_coordinator)
-    Process.exit(group_coordinator, :kill)
-
-    receive do
-      {:DOWN, ^ref, _, _, _} -> :ok
-    end
-
+  def disconnect(group_coordinator, client_id) do
+    :ok = :brod_group_coordinator.stop(group_coordinator)
```

Comment:
@FedericoPanS please revert this change and maintain the old shutdown as this would require brod 3.15.2 or later.